kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/ProxyDeployment.java

```diff
@@ -77,10 +79,15 @@ static Map<String, String> podLabels() {
     }
 
     private PodTemplateSpec podTemplate(KafkaProxy primary) {
+        Map<String, String> labelsFromSpec = Optional.ofNullable(primary.getSpec().getPodTemplate())
+                .map(PodTemplateSpec::getMetadata)
+                .map(ObjectMeta::getLabels)
+                .orElse(Map.of());
         // @formatter:off
         return new PodTemplateSpecBuilder()
```

Comment:
If you're able to use the raw extension idea with the CEL validation, then a better way of doing this might be

```java
new PodTemplateSpecBuilder(primary.getSpec().getPodTemplate())...
```

literally copying the user's template, and then editing-in our adjustments.

Reply:
That's how I started out, I figured that the CRD is preventing any other fields being set so it was safe.

I took @SamBarker's comments to be that we should be more defensive in the operator, even if the CRD currently does not let the user specify anything but labels.

Reply:
I thought the plan was to expose the full pod schema and validate and fail/warn if more than labeles were specified so that we remained in control of what was being generated.